.github/workflows/publish.yml

```diff
@@ -8,10 +8,15 @@ jobs:
       image: eclipse-temurin:21
       options: --user root
     steps:
+      - name: Install git
+        run: apt-get update && apt-get install -y git
+      - name: Configure git for container
+        run: git config --global --add safe.directory "$(pwd)"
+
       - uses: actions/checkout@v4
```

Comment:
I think you can skip installing, configuring git and fetching the tags manually by using the [`fetch-tags: true` option for `actions/checkout`](https://github.com/actions/checkout/tree/v4?tab=readme-ov-file#usage) (though maybe the install step is required to get the build number, but the readme for the action does show [an example where its not needed](https://github.com/actions/checkout/tree/v4?tab=readme-ov-file#usage), maybe it has to do with the container image we're using)